test/polygon_fuzz.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <atomic>
+#include <future>
+
+#include "fuzztest/fuzztest.h"
+#include "gtest/gtest.h"
+#include "polygon.h"
+
+using namespace fuzztest;
+
+void TriangulationNoCrash(
+    std::vector<std::vector<std::pair<float, float>>> input, float precision) {
+  if (precision < 0) precision = -1;
+  manifold::PolygonParams().intermediateChecks = true;
+  manifold::Polygons polys;
+  size_t size = 0;
+  for (const auto &simplePoly : input) {
+    polys.emplace_back();
+    for (const auto &p : simplePoly) {
+      polys.back().emplace_back(p.first, p.second);
+      size++;
+    }
+  }
+  std::atomic<pid_t> tid;
+  std::atomic<bool> faulted(true);
+  auto asyncFuture =
+      std::async(std::launch::async, [&polys, &faulted, &tid, precision]() {
+        tid.store(gettid());
+        try {
+          manifold::Triangulate(polys, precision);
+          faulted.store(false);
+        } catch (manifold::geometryErr e) {
+          // geometryErr is fine
+          faulted.store(false);
+        } catch (...) {
+          printf("got unexpected error\n");
+        }
+      });
+  if (asyncFuture.wait_for(std::chrono::milliseconds(
```

Comment:
What is the async for? Is this so we can fuzz all the test cases in parallel? Perhaps if we reduce to just a few fuzzing test cases this won't be necessary anymore?

I also found `--gunit_fuzz_for=10m` in other documentation - would this help?

I haven't found *much* more documentation on this internally, but there are people I can ask. I'm not very familiar with any of this yet, but if you want to pass me a fairly specific question to ask, I can try to hunt down an answer.

Reply:
No, this async is for implementing timeout. From their code, it seems that the `--gunit_fuzz_for` arg is used for `Runs all fuzz tests in fuzzing mode for the specified duration`, i.e. terminate after the specified duration instead of test timeout.

Parallel fuzzing requires support from their engine, I don't think I see anything related in their code for now.

Maybe you can ask about these two things: test timeout and parallel fuzzing.

Reply:
What is the difference between a test timeout and terminating a test after a specified duration?

Reply:
Test timeout is for individual test, while the option is for the whole fuzzing test.